File: peuler/ex14/ex14.cpp
#include<iostream>

unsigned long int collatzLength(unsigned long int n) {
  unsigned long int length = 0;
  while (n != 1)
	if (n % 2 == 0) {
	  n = (unsigned long int) n / 2;
	  length += 1;
	}
	else {
	  n = 3 * n + 1;
	  length += 1;
	}
  return length;
}

int main() {
  unsigned long int start, thisOne;
  unsigned long int maxLength = 0;

  for (start = 1000000; start > 1; start--)
	if (maxLength < collatzLength(start)) {
	  maxLength = collatzLength(start);
	  thisOne = start;
	}

  printf("%lu is the longest Collatz sequence starting at %lu\n", maxLength, thisOne);
  
  return 0;
}

File: peuler/ex14/ex14.py
def collatzLength(n):
    length = 0
    while (n != 1):
        if (n % 2 == 0):
            n = int(n / 2)
            length += 1
        else:
            n = 3 * n + 1
            length += 1
    return length

maxLength = 0
thisOne = 1
for start in range(1000000, 1, -1):
    if (maxLength < collatzLength(start)):
        maxLength = collatzLength(start)
        thisOne = start

print(str(maxLength) + " is the largest Collatz sequence starting at " + str(thisOne))
        
